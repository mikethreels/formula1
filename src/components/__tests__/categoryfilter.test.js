/* eslint-disable max-len */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import CategoryFilter from '../categoryfilter';
import Reducers from '../../reducers/index';

const scores = {
  position: 3,
  points: 10,
  wins: 9,
  Driver: {
    code: 'Ver',
    givenName: 'Max',
    familyName: 'Verstappen',
    nationality: 'Netherlands',
    permanentNumber: 33,
    dateOfBirth: '01-01-1997',
    url: 'https://testurl.com',
  },
  Constructors: [
    {
      url: 'https://testurl.com',
      name: 'Red Bull',

    },
  ],
};

const state = {
  drivers: [],
  scores: [scores],
  filter: '',
};

const store = createStore(
  Reducers,
  state,
);

it('renders Category filter correctly', () => {
  const testFunction = () => {};

  const stock = {
    categories: ['Mercedes', 'Red-bull', 'Ferrari'],
    callback: testFunction,
    value: 'All',
  };

  const tree = renderer.create(
    <Provider store={store}>
      <CategoryFilter categories={stock.categories} callback={stock.callback} value={stock.value} />
    </Provider>,
  );

  expect(tree).toMatchSnapshot();
});
