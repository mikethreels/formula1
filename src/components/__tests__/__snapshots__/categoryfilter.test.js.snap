// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Category filter correctly 1`] = `
<div>
  <label
    htmlFor="category"
  >
    DRIVER FILTER: 
  </label>
  <select
    id="category"
    onChange={[Function]}
    value="All"
  >
    <option
      selected={true}
    >
      All
    </option>
    <option>
      Mercedes
    </option>
    <option>
      Red-bull
    </option>
    <option>
      Ferrari
    </option>
  </select>
</div>
`;
